bindings/python/src/pgvecto_rs/highapi/client.py

```diff
@@ -0,0 +1,199 @@
+from typing import List, Literal, Optional, Tuple, Type, Union
+from uuid import UUID
+
+from numpy import ndarray
+from sqlalchemy import (
+    ColumnElement,
+    Float,
+    String,
+    create_engine,
+    delete,
+    insert,
+    select,
+    text,
+)
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.engine import Engine
+from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm.session import Session
+
+from pgvecto_rs.sqlalchemy import Vector
+
+from .embedder import BaseEmbbeder
+from .filter import FilterFunc
+from .record import Record, RecordORM, RecordORMType
+
+
+class Client:
+    _engine: Engine
+    _table: Type[RecordORM]
+    embedder: Optional[BaseEmbbeder]
+    dimension: int
+
+    # ================ Initialization ================
+    def __init__(
+        self,
+        db_url: str,
+        table_name: str,
+        dimension: int,
+        embedder: Optional[BaseEmbbeder] = None,
+        new_table: bool = False,
+    ) -> None:
+        """Connect to an existing table or create a new one.
+
+        Args:
+            db_url (str): url to the database.
+            table_name (str): name of the table.
+            dimension (int): dimension of the embeddings.
+            embedder (Optional[BaseEmbbeder], optional): Defaults to None.
+            new_table (bool, optional): Defaults to False.
+
+        Raises:
+            ValueError: When the dimension of the embedder does not match the given dimension.
+        """
+
+        class _Table(RecordORM):
+            __tablename__ = f"table_{table_name}"
+            id: Mapped[UUID] = mapped_column(
+                postgresql.UUID(as_uuid=True), primary_key=True
+            )
+            text: Mapped[str] = mapped_column(String)
+            meta: Mapped[dict] = mapped_column(postgresql.JSONB)
+            embedding: Mapped[ndarray] = mapped_column(Vector(dimension))
```

Comment:
I don't get it. Why do we need to inherit `RecordORM` then override all the attributes?

Reply:
It's just how the SQLalchemy works...